### PR TITLE
Add Commit Message Linting

### PR DESCRIPTION
**Pull request by solomonhawk**
- Pulls in https://github.com/conventional-changelog/commitlint with the conventional config
- Wires up husky git hook

### CODE REVIEW

File: .commitlintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
+  rules: {
+    "type-enum": [2, "always", ['feat', 'fix', 'docs', 'style', 'refactor', 'test', 'revert', 'build', 'chore', 'ci']]
+  }
+}
```
Comment (olets) [inline .commitlintrc.cjs:6]: @commitlint/config-conventional enumerates types, so you'll get the friendly "must be one of" error without adding the type-enum here

Comment (olets) [reply .commitlintrc.cjs:6]: (late comment — opened the link from slack, but didn't look at it until now:)

Comment (solomonhawk) [reply .commitlintrc.cjs:6]: I included this because I wanted to customize/extend the allowed types :)

Comment (olets) [reply .commitlintrc.cjs:6]: gotcha cool. so (just chatting now) is the opinion that "perf" should instead be rolled into … maybe "feat"?

Comment (olets) [reply .commitlintrc.cjs:6]: oh oh I see https://commitlint.js.org/#/reference-rules?id=type-enum lies. the fact that your last three come after revert makes me think you may have been fooled by it. https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional

**Review by dlederle — COMMENTED** [review]
wooooah, neat.

File: .commitlintrc.cjs (hunk 1 above)
Comment (solomonhawk) [reply .commitlintrc.cjs:6]: OOOO yep, this got me. I think I can just remove it unless I have any additional types I want to add. Thanks!

**Pull request merged**